### ISSUE TITLE
Crash on ARM64 Android

### ISSUE THREAD

**Issue by zrose584** [WeiJiLab/kernel-hook-framework#4]
Hi, there seems to be a general problem with "unprotecting" the kernel regions / making them writable on newer android kernels. At least it is mentioned at different places, e.g.:

[Stackoverflow Android 12 kernel 4.14.190](https://stackoverflow.com/questions/61247838/cannot-use-set-memory-rw-in-linux-kernel-on-arm64?noredirect=1&lq=1#comment131132243_61465861)
https://github.com/m0nad/Diamorphine/issues/37

The [dmesg.log](https://github.com/WeiJiLab/kernel-hook-framework/files/12612874/dmesg.log)

I am using 4.14 (had to make a few adjustments to the code).
On another phone with kernel 4.4 I had no issue hijacking some kernel symbols.

**Comment by liutgnu** [comment]
Hi, Thanks for reporting the issue.

Do you know if the issue can be reproduced on other linux distros(fedora/debian/ubuntu etc) other than android?

Frankly I didn't follow the android development recently, so this project hasn't been tested on android for quite sometime. And I doesn't get a android 12 phone at hand and it would be time consuming to install a android emulator...

Anyway, I will look into the issue when I have time. Thanks!

**Comment by zrose584** [comment]
> Do you know if the issue can be reproduced on other linux distros(fedora/debian/ubuntu etc) other than android?

No, I don't know. If you have powerful-enough arm64 hardware, maybe this can be reproduced using genymotion - they have some arm64 images: https://support.genymotion.com/hc/en-us/articles/9511224511901-Which-Android-versions-are-available-

> And I doesn't get a android 12 phone

Fwiw, Kernel 4.14 seems to be used as early as Android 10 in some devices

The project [bmax121/KernelPatch](https://github.com/bmax121/KernelPatch) looks interesting, but I am not sure if it uses a (new) way to disable this "additional protection", or it just patches the kernel memory before those additional protections got activated (the project patches the kernel image and hooks into the very early init phases)

**Comment by liutgnu** [comment]
It has been a long time, finally I got a android VM of newer kernel to test, and I did fix a stack safety check issue. However I didn't reproduce the exact same issue as reported this one. So I assume the issue is no longer reproducable and close it. Feel free to reopen if encounter again.
